templates/base/client/.rtag/types.ts.hbs

```diff
@@ -5,6 +5,22 @@ export enum {{@key}} {
   {{label}},
   {{/each}}
 }
+{{else if (eq type "union")}}
+{{#each options}}
+interface {{@../key}}_{{> renderArg}} {
+  type: "{{> renderArg}}"
+  value: {{> renderArg}}
+}
+{{/each}}
```

Comment:
why is this necessary?

Reply:
we need the "type" string literal in order to write the type guards. another option that just occurred to me is we could write these as

`type Fruit = (Apple & {type: "Apple"}) | (Orange & {type: "Orange"}` which might be slightly friendlier. what do you think?